tests/integrators/advection_01.cc

```diff
@@ -63,7 +63,7 @@ test_cell(const FEValuesBase<dim> &fev)
         u    = 0.;
         u(i) = 1.;
         w    = 0.;
-        fev.get_function_values(u, indices, uval, true);
+        fev.get_function_values(u, indices, make_array_view(uval), true);
```

Comment:
Are the changes in these integrators the only places that need to be touched? Do you happen to know why here? Is it because we have `std::vector<std::vector<double>>` that can somehow not be automatically converted? I think that might be something we need to describe more clearly in the incompatibilities section of a changelog?

Reply:
It's because the template instantiation is different: since we instantiate on `Number` instead of `VectorType` the deduction process is slightly different and we need to explicitly pass an ArrayView (instead of relying on the implicit conversion constructor). I added a changelog entry summarizing the problem.

I reran all the tests which use FEValues or FEFaceValues locally and they all pass so I think I fixed all the relevant ones in this PR.